content/fr/articles/js/observable/observable.part1.md

```diff
@@ -0,0 +1,210 @@
+
+En NodeJS, voici comment se présente la fonction qui nous permet de lire un fichier : `fs.readFile(path, options, callback)`
+
+*elle nous renvoie soit une erreur, soit le contenu du fichier: (err, data)*
+
+* Implémentons nos trois fonctions de rappel :
+
+``` Javascript
+// callback
+
+function afficheData(data) {
+    console.log(value)
+}
+
+function afficheErreur(data) {
+    console.log(erreur)
+}
+
+function afficheLectureComplete() {
+    console.log(‘Lecture complète du fichier’)
+}
+```
+
+* Appliquons nos fonctions de rappel à fs.readFile:
+
+>On imagine un fichier nommé alphabet.txt, qui comporte toutes les lettres de l’alphabet
+
+``` Javascript
+fs.readFile('./alphabet.txt', {encoding: 'utf-8'}, (err, data) => {
+    if (err) {
+        afficheErreur(err)
+    } else {
+        afficheData(data)
+    }
+    afficheLectureComplete()
+})
+```
+
+Tout fonctionne. Mais il y a un problème.. écrire des lignes et des lignes de fonctions de rappel tout le temps devient très vite laissant et épuisant pour les pauvres développeur paresseux que nous sommes. Par conséquent pourquoi ne pas imaginer d’implémenter une petite API / Object / Bloc de fonction qui pourrait s’adapter à tous formes de donner synchrones/asynchrones ( fournies par arrayOfValues, fetch, fs.readFile etc… ).
+
+```Javascript
+
+// API / Object / Ensemble de fonctions de rappel qui nous permet de traiter toutes les formes de données
+
+const observer = {
+   next: function(val) { console.log(val) },
+   error: function(err) { console.log(err) },
+   complete: function() { console.log(‘complete!’)
+}
+```
+
+Maintenant nous pouvons juste envoyer notre objet observer en tant que “callback” :
+
+`function subscribe(observer) {
+  [1,2,3,4,5].forEach(observer.next)
+}`
+
+Le nommage de cette fonction n'est pas du au hasard. "Subscribe" par soucis de comprehension. Finalement, nous pouvons très imaginer que nous souscrivons à une source donnée qui est interprétée dans notre “API” de callback **observer**.
+
+> Nous disposons d’une source de données. Pour avoir accès à cette source de données, nous allons souscrire.
+
+Maintenant imaginons un monde où toutes les sources de données synchrones comme asynchrones nous permettent un accès à ses données tout simplement par souscription. Nous allons appelé ces objets des Observable.
+
+**Un observable est donc un objet qui comporte une méthode subscribe qui prend
+prend en paramètre une suite de callback(Observer).
+L'observable s'occupe d'appeler l'observer adéquat en fonction des situations**
+
+* Deux exemples de ce monde parfait avec une suite d’observable :
+
+```Javascript
+
+const observer = {
+  next: function(val) { console.log(val) },
+  err: function(err) { console.log(err) },
+  complete: function() { console.log('Done.') }
+}
+
+const observableWithArrayOfInteger = {
+   values: [1,2,'trois',4,5],
```

Comment:
quel est l'intérêt de faire un objet avec accès à `this.values` plutôt que d'en faire un paramètre? pas très clair